security/self-assessment.md

```diff
@@ -0,0 +1,363 @@
+that is also running on Kubernetes. We believe that Kubernetes Native
+Infrastructure, or managing your infrastructure just like your applications, is
+a powerful next step in the evolution of infrastructure management.
+
+The Metal3 project is also building integration with the Kubernetes cluster-api
+project, allowing Metal3 to be used as an infrastructure backend for Machine
+objects from the Cluster API. These components integrate seamlessly to leverage
+the Kubernetes ecosystem and automate the provisioning and management of
+bare-metal infrastructure.
+
+### Non-goals
+
+Metal3's non-goals include:
+
+* Non-baremetal hardware provisioning
+* Direct hardware management without Ironic
+* Operating system configuration management after provisioning
+* Network provisioning or SDN implementation
+* Providing a standalone solution outside of Kubernetes
+* Exposing full set of Ironic features
+
+## Self-assessment use
+
+This self-assessment is created by the Metal3 team to perform an internal
+analysis of the project's security.  It is not intended to provide a security
+audit of Metal3, or function as an independent assessment or attestation of
+Metal3's security health.
+
+This document serves to provide Metal3 users with an initial understanding of
+Metal3's security, where to find existing security documentation, Metal3
+plans for security, and general overview of Metal3 security practices, both
+for development of Metal3 as well as security of Metal3.
+
+This document provides the CNCF TAG-Security with an initial understanding of
+Metal3 to assist in a joint-assessment, necessary for projects under
+incubation.  Taken together, this document and the joint-assessment serve as a
+cornerstone for if and when Metal3 seeks graduation and is preparing for a
+security audit.
+
+## Security functions and features
+
+**Critical**
+
+* Ironic: Ironic is the ultimate component which provisions the bare metal
+   hosts, talks to the real hardware and is able to perform very destructive or
+   malicious actions.
+
+**Security Relevant**
+
+* Ironic Python Agent (IPA): Ironic's agent counterpart that is responsible for
+   doing initial bootstrapping of the node.
+* Bare Metal Operator (BMO): Currently BMO has cluster-wide K8s Secret access
+   so it can read BMC secrets from any namespace. This is needed as access to
+   BareMetalHost CRDs which are separated from the secrets, to enable role
+   separation for the users and admins.
+
+## Project compliance
+
+The Metal3 project does not comply with any specific security standard.
+
+## Secure development practices
+
+**Development Pipeline**
+
+* We run many linters and unit tests first, then e2e suite as required. If
+   PR is touching a major feature, we run optional e2e feature suite. Most
+   testing is configured via
+   [Prow config](https://github.com/metal3-io/project-infra/blob/main/prow/manifests/overlays/metal3/config.yaml)
```

Comment:
Link needs updating (sorry :see_no_evil: )

Reply:
Good catch, I'll update.

Reply:
Done in next push.